### DISCUSSION TITLE
dropouts on snapcast clients

### DISCUSSION THREAD

**Discussion by jampr21** [snapcast/snapcast#817] [Q&A]
Hi badaix, 
regarding https://github.com/badaix/snapcast/issues/774#issuecomment-785704004
It might be, that it is really an issue with wifi. But it is still a bit unclear for me. One client is raspi4 connected via LAN (same subnet) and a raspi4zeroW connected via wifi (different subnet). If the wifi latency is to high on wifi, will it also cause also dropouts on the client connected via LAN? (now everything on 0.24)
thanks a lot for your support

**Comment by Taxicletter** [comment]
I have a problem with dropouts as well.
I use SnapCast Server on Kodi (latest, 18) on RPI3, and HifiBerryOS (latest, 20210203), also on a RPI3 (with Amp).
A few months ago this worked perfectly, but now there are huge dropouts, in fact more dropout then actual playing.
Both Raspberry's are on the same network. There's not much traffic on that network.

**Reply by kingosticks** [reply]
You need to specify what version of snapcast you are using and if it's not the latest version then you need to update it first.

**Reply by Taxicletter** [reply]
On my Kodi the snapcast version is 9.2.0.102. If I click on "update" it appears to be the latest version.

**Reply by jampr21** [reply]
what helped me was to set the swappiness to 0
`sysctl -w vm.swappiness=0`

**Reply by jampr21** [reply]
Second adjustment I did for wiping out the last hickups:
raised the fragments and lowered the buffer a bit in /etc/default/snapclient.conf
`SNAPCLIENT_OPTS="-h 192.168.0.22 -s plughw:CARD=AUDIO  --player alsa:buffer_time=10,fragments=8"`

**Comment by teslasparks301** [comment]
I had terrible latency problems using wifi.  I tried messing with client and server parameters to no avail.  Then I noticed my wifi ( a 'hipower' usb wifi dongle ) connected at about 84 Mb/s on the 2.4 band. This had to handle the traffic from the smb share (or webradio site), play the music, and then send it to the snapclients, one of which is on the snapserver box itself.  Plus other miscellaneous network traffic.  I knew I should be able to get a faster connection than 84 Mb/s.  My USB dongles are old because my network speed needs are usually modest. In this case I bought a new dongle capable of 8xx Mb/s.  Being a cheapskate, bought one of the cheapest on Amazon. Nothing very special except that it has dual antennae. I switched to the 5 GHz band and am able to connect at 400 Mb/s.  And this from being enclosed in an entertainment cabinet.  Things are working quite nicely.  Music is in sync and smooth.

Now the software/hardware.  My clients use snapOS on rpi zero and use the on-board wifi.  The sound hat is a hifiberry zero. The server is a rpi 3B+, hifiberry dac+, and the usb wifi dongle described above. The software is rAudio-1.  The snapserver is built-in. I am not being terribly specific about the software/hardware because the big difference was getting a fast wifi connection on the server and not placing the clients all the way across the house from the access point.  If you need to have that placement, new wifi dongles may be needed for the clients.

When buying dongles, being described as linux compatible may mean you have to procure the source and compile it yourself.  This can be a challenge on dedicated music player images where the file system tends to be very slim in resources. Like build tools and kernel headers.  Otherwise, one will have to cross-compile.  Easiest is to get a dongle that already has the driver in the module tree.  That may be easier said than done.

This is my story. The problem was clearly in the server connection and not the clients.  YMMV.

mike

**Reply by Taxicletter** [reply]
Thanks for this information!
I'm currently not using SnapCast, but I'll store this for later use.
